SU2_CFD/include/objectives/turbomachinery/CTurbomachineryPerformance.hpp

```diff
@@ -0,0 +1,256 @@
+                              unsigned short nDim,
+                              su2double areaIn,
+                              su2double radiusIn,
+                              su2double areaOut,
+                              su2double radiusOut);
+
+  void ComputePerformance(const CTurbomachineryCombinedPrimitiveStates &primitives) override;
+
+};
+
+class CPropellorBladePerformance : public CTurbomachineryBladePerformance {
+
+  public:
+  CPropellorBladePerformance(CFluidModel& fluidModel,
+                              unsigned short nDim,
+                              su2double areaIn,
+                              su2double radiusIn,
+                              su2double areaOut,
+                              su2double radiusOut);
+
+  void ComputePerformance(const CTurbomachineryCombinedPrimitiveStates &primitives) override;
+
+};
+
+class CTurbomachineryStagePerformance {
+   protected: 
+      su2double TotalStaticEfficiency, TotalTotalEfficiency, KineticEnergyLoss, TotalPressureLoss, EntropyGen, PressureRatio, EulerianWork;
+      CFluidModel &fluidModel;
+   public:
+      CTurbomachineryStagePerformance(CFluidModel& fluid);
+      ~CTurbomachineryStagePerformance();
+      virtual void ComputePerformanceStage(CTurbomachineryState InState, CTurbomachineryState OutState, CConfig* config);
+      virtual void ComputeTurbineStagePerformance(CTurbomachineryState InState, CTurbomachineryState OutState);
+      virtual void ComputeCompressorStagePerformance(CTurbomachineryState InState, CTurbomachineryState OutState);
+      su2double GetTotalStaticEfficiency() const { return TotalStaticEfficiency; }
+      su2double GetTotalTotalEfficiency() const { return TotalTotalEfficiency; }
+      su2double GetEulerianWork() const { return EulerianWork; }
+      su2double GetEntropyGen() const { return EntropyGen; }
+      su2double GetPressureRatio() const { return PressureRatio; }
+      su2double GetKineticEnergyLoss() const { return KineticEnergyLoss; }
+};
+
+class CTurbomachineryPerformance {
+  private:
+  vector <vector<shared_ptr < CTurbomachineryBladePerformance>>>
+  BladesPerformances;
+
+  static void ComputePerBlade(vector <shared_ptr<CTurbomachineryBladePerformance>> const bladePerformances,
+                              vector <CTurbomachineryCombinedPrimitiveStates> const bladePrimitives);
+
+  static void ComputePerSpan(shared_ptr <CTurbomachineryBladePerformance> const spanPerformances,
+                             const CTurbomachineryCombinedPrimitiveStates &spanPrimitives);
+  // vector<shared_ptr<CTurbomachineryStagePerformance>> StagePerformances;
+  // shared_ptr<CTurbomachineryStagePerformance> MachinePerformances;
+  public:
+  CTurbomachineryPerformance(const CConfig& config, const CGeometry& geometry, CFluidModel& fluidModel);
+
+  vector <vector<shared_ptr < CTurbomachineryBladePerformance>>>
+
+  GetBladesPerformances() const { return BladesPerformances; }
```

Comment:
please format this a little more clearly. Since these are new files our current guidelines are to run clang-format (you run it from the root of the repo which is where we have the file with the code style).